src/libasr/codegen/asr_to_c_cpp.h

```diff
@@ -549,7 +549,11 @@ R"(#include <stdio.h>
         if (ASR::is_a<ASR::Var_t>(*x.m_target)) {
             visit_Var(*ASR::down_cast<ASR::Var_t>(x.m_target));
             target = src;
-            if( ASRUtils::is_array(ASRUtils::expr_type(x.m_target)) && !is_c ) {
+            // auto v = ASRUtils::EXPR2VAR(x.m_target);
+            // if (is_c && ASRUtils::is_character(*v->m_type)) {
+            //     target = "free(" + target + "); " + target;
+            // }
```

Comment:
Using `free` here fails few tests. 
```bash
90% tests passed, 3 tests failed out of 29

Label Time Summary:
c          =   0.71 sec*proc (29 tests)
cpython    =   0.70 sec*proc (21 tests)
llvm       =   0.71 sec*proc (29 tests)

Total Test time (real) =   0.73 sec

The following tests FAILED:
         10 - print_01 (Subprocess aborted)
         18 - loop_01 (Subprocess aborted)
         21 - test_c_interop_02 (Subprocess aborted)
```

These tests have string allocated in `stack memory` and I guess calling `free` on `stack memory` is causing the failure.